registry/registry.go

```diff
@@ -723,11 +737,89 @@ func tarReader(reader io.Reader, contentType string) (*tar.Reader, error) {
 	return tar.NewReader(reader), nil
 }
 
+// CheckVersion controls the matching versions between retrieved tarball and
+// options
+func (t *Tarball) CheckVersion(expectedVersion string) (bool, error) {
+	tbManifestVersion := t.Manifest.Version
+	tbPackageVersion := t.PackageVersion
+
+	var match bool
+	var errm error
+
+	if tbManifestVersion != "" {
+		match = VersionMatch(expectedVersion, tbManifestVersion)
+	}
+	if !match {
+		errm = multierror.Append(errm,
+			fmt.Errorf("%q field does not match (%q != %q)",
+				"version", expectedVersion, tbManifestVersion))
+	}
+
+	if tbPackageVersion != "" {
+		match = VersionMatch(expectedVersion, tbPackageVersion)
```

Comment:
Should I understand that the manifest version is mandatory but the package version is optional?

Reply:
Yes, I kept the legacy checks https://github.com/cozy/cozy-apps-registry/blob/master/registry/registry.go#L809. We do not check the package version if `package.json` is not present in the tarball